packages/remix-react/__tests__/hook-types-test.tsx

```diff
@@ -1,78 +1,83 @@
-import type { TypedResponse, UseDataFunctionReturn } from "../components";
+import type { TypedResponse } from "../serialize";
+import type { useLoaderData } from "../components";
 
 function isEqual<A, B>(
   arg: A extends B ? (B extends A ? true : false) : false
 ): void {}
 
+// not sure why `eslint` thinks the `T` generic is not used...
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type LoaderData<T> = ReturnType<typeof useLoaderData<T>>;
```

Comment:
note: this test is for testing `useLoaderData`'s types, so we actually grab types from `useLoaderData` instead of assuming its using `UseDataFunctionReturn` or `SerializeFrom`.